From 5c67cf2774a8b964f4d7cd92a4c447da81ac6087 Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Fri, 12 Sep 2014 16:22:44 +0200
Subject: [PATCH] udev: don't close std{in,out,err}

Rather than printing debug output to stderr and redirecting this to /dev/null when not wanted,
use the correct log_*() function in the first place.
---
 src/udev/udevd.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 8922ff9f8e..e90d9dacb0 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -1221,18 +1221,6 @@ int main(int argc, char *argv[]) {
 
         log_info("starting version " VERSION "\n");
 
-        if (!arg_debug) {
-                int fd;
-
-                fd = open("/dev/null", O_RDWR);
-                if (fd >= 0) {
-                        dup2(fd, STDIN_FILENO);
-                        dup2(fd, STDOUT_FILENO);
-                        dup2(fd, STDERR_FILENO);
-                        close(fd);
-                }
-        }
-
         fd_inotify = udev_watch_init(udev);
         if (fd_inotify < 0) {
                 log_error("error initializing inotify");
